### PR TITLE
integration-tests: add abort tests

### PR DESCRIPTION
**Pull request by fgimenez**
Added tests according to the recommendations from the Ubuntu Personal - Product Test team. A reference to the Test Case ID is included in each test comment.

The suite has been added to autopkgtest.

### CODE REVIEW

File: debian/tests/integrationtests [hunk 1]
```diff
@@ -33,7 +33,7 @@ if [ -z "$ADT_REBOOT_MARK" ]; then
     go test -tags classiconly -c ./integration-tests/tests
 fi
 
-./tests.test  -check.vv -check.f 'snapHelloWorldExampleSuite|snapOpSuite|searchSuite|installSuite|authSuite|networkInterfaceSuite|networkBindInterfaceSuite|homeInterfaceSuite|logObserveInterfaceSuite|unity7|snapPersistsSuite'
+./tests.test  -check.vv -check.f 'snapHelloWorldExampleSuite|snapOpSuite|searchSuite|installSuite|authSuite|networkInterfaceSuite|networkBindInterfaceSuite|homeInterfaceSuite|logObserveInterfaceSuite|unity7|snapPersistsSuite|abortSuite'
```
Comment (niemeyer) [inline debian/tests/integrationtests]: This is getting a bit wild. In a follow up it'd be good to clean it up somehow.

**Comment by niemeyer** [comment]
Integration tests are failing.

**Comment by fgimenez** [comment]
retest this please

**Comment by niemeyer** [comment]
retest this please

**Comment by niemeyer** [comment]
retest this please

**Comment by niemeyer** [comment]
retest this please

**Comment by niemeyer** [comment]
retest this please

**Comment by fgimenez** [comment]
@niemeyer pls do not merge no matter the test results, the wait.ForCommand function seems to be misbehaving when a snap command is given, i'll fix it tmw. Thx! :)

File: debian/tests/integrationtests (hunk 1 above)
Comment (fgimenez) [inline debian/tests/integrationtests]: Indeed, we have been talking about this already and will be addressed shortly cc/ @elopio

**Comment by fgimenez** [comment]
All should be fixed now, the problem was related to the function adding the coverage options in the cli package and is already resolved, ready for review
retest this please

**Comment by mvo5** [comment]
:+1:

**Comment by niemeyer** [comment]
There were conflicts which prevented it from being merged earlier, so the imports will need to be fixed now that the code has moved.

**Comment by fgimenez** [comment]
@niemeyer done, i've taken the opportunity to update the CI jobs to point to this repo and they are correctly triggered, but we also need to add the bot (snappy-m-o as regular user, not admin) to the new org in order to see the status updated here.

Thanks!

**Comment by fgimenez** [comment]
retest this please

**Comment by niemeyer** [comment]
LGTM, but there are some conflicts.

File: integration-tests/testutils/cli/cli_test.go [hunk 2]
```diff
@@ -420,6 +420,20 @@ func (s *cliTestSuite) TestAddOptionsToSnappyCommand(c *check.C) {
 	}
 }
 
+func (s *cliTestSuite) TestAddOptionsDoesNotModifyOriginalCmds(c *check.C) {
```
Comment (elopio) [inline integration-tests/testutils/cli/cli_test.go]: nice! so we can now print only the original cmd?

**Comment by elopio** [comment]
:+1:

File: integration-tests/testutils/cli/cli_test.go (hunk 2 above)
Comment (fgimenez) [inline integration-tests/testutils/cli/cli_test.go]: Nope, there was a problem with the way the coverage options were inserted and the original input `cmds` slice was modified. The problem has arisen after feeding a snap command to `wait.ForCommand`, when the same cmds slice is called more than once.

It's easy to remove the coverage options from the command we print, I can propose that in another branch. IMO we should wait to see how goes the upcoming verbosity flag branch.

**Comment by fgimenez** [comment]
> LGTM, but there are some conflicts.

@niemeyer Thanks, fixed

**Pull request merged**